### PR TITLE
[ERRATA] [V7-1539] Change default behaviour in ClassificationDataset

### PR DESCRIPTION
**Pull request by FrancescoSaverioZuppichini**
- Better naming, before `target` was being overwritten.
- `assert` if no `tags`
- default behaviour is to the first tag, if `multi_label` one-hot them

### CODE REVIEW

File: darwin/torch/dataset.py [hunk 1]
```diff
@@ -115,18 +115,20 @@ def get_target(self, index: int):
         Returns the classification target
         """
 
-        target = self.parse_json(index)
-        annotations = target.pop("annotations")
+        data = self.parse_json(index)
+        annotations = data.pop("annotations")
         tags = [a["name"] for a in annotations if "tag" in a]
 
+        assert len(tags) >= 1, f"No tags were found for index={index}"
```
Comment (Fl4m3Ph03n1x) [inline darwin/torch/dataset.py:122]: I am confused. If `tags` has content, does it not mean we actually did find tags for the index?
Also, I can't really remember regarding the use of asserts in functions, we did mention this topic last week. 
@andreaazzini  sorry to bug you, but can you pop in and remind us of what the conclusion was? Were we OK with it?

**Review by Fl4m3Ph03n1x — CHANGES_REQUESTED** [review]
I have a small question. Other than that, would it be feasible to add a test?

File: darwin/torch/dataset.py (hunk 1 above)
Comment (FrancescoSaverioZuppichini) [reply darwin/torch/dataset.py:122]: assert will raise an exception if `len(tags) >=1` is `False`, so no tag were found

Comment (Fl4m3Ph03n1x) [reply darwin/torch/dataset.py:122]: Good point !

**Review by Fl4m3Ph03n1x — APPROVED** [review]

**Pull request merged**